#!/usr/bin/env python
PACKAGE = "vwbot_controller"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator() 

gen.add( "param_pid_action1_p", double_t,  0,     "PID_P",   1.7,      0,       10)
gen.add( "param_pid_action1_i", double_t,  0,     "PID_I",   0.0,      0,       10)
gen.add( "param_pid_action1_d", double_t,  0,     "PID_D",   0.55,      0,       10)
gen.add( "param_pid_action1_vel", double_t,  0,     "PID_Vel",   0.30,      0,       10)

gen.add( "change_1_x", double_t,  0,     "The point1 to change",   1.5,      0,       5)
gen.add( "change_1_y", double_t,  0,     "The point1 to change",   -1.0,      0,      5)
gen.add( "change_2_x", double_t,  0,     "The point2 to change",   1.5,      0,       5)
gen.add( "change_2_y", double_t,  0,     "The point2 to change",   0.0,      0,       5)
gen.add( "change_3_x", double_t,  0,     "The point3 to change",   1.5,      0,       5)
gen.add( "change_3_y", double_t,  0,     "The point3 to change",   1.0,      0,       5)

gen.add( "point_yellow_x", double_t,  0,     "The yellow point",   3.30,      0,       5)
gen.add( "point_yellow_y", double_t,  0,     "The yellow point",   0.55,      0,       5)
gen.add( "point_red_x", double_t,  0,     "The red point",   3.30,      0,       5)
gen.add( "point_red_y", double_t,  0,     "The red point",   0.00,      0,       5)
gen.add( "point_blue_x", double_t,  0,     "The blue point",   3.30,      0,       5)
gen.add( "point_blue_y", double_t,  0,     "The blue point",   -0.55,      0,      5)

gen.add( "dis_length_when_catch", double_t,  0,     "The aim length to catch",   0.14,      0,       1)
gen.add( "wait_the_task_has_ball3", double_t,  0,     "Wait the task_has_ball3",   0,      0,       10)

gen.add( "absolute_yaw_to_ball", double_t,  0,     "absolute_yaw_to_ball",   15,      0,       30)
gen.add( "pose_variation", double_t,  0,     "PoseVariation",   0.035,      0,       1)

exit(gen.generate(PACKAGE, "custom_dynamic", "Tutorials"))

